lib/AppInfo/Application.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * @author Jörn Friedrich Dreyer <jfd@butonic.de>
+ *
+ * @copyright Copyright (c) 2019, ownCloud GmbH
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\User_LDAP\AppInfo;
+
+use OCA\User_LDAP\Group_Proxy;
+use OCA\User_LDAP\Helper;
+use OCA\User_LDAP\ILDAPWrapper;
+use OCA\User_LDAP\LDAP;
+use OCA\User_LDAP\User_Proxy;
+
+class Application extends \OCP\AppFramework\App {
+	/**
+	 * @param array $urlParams
+	 */
+	public function __construct($urlParams = []) {
+		parent::__construct('user_ldap', $urlParams);
+		$this->registerService();
+		$this->checkCompatability();
```

Comment:
just checking, this is the usual thing to be done once, when the app is enabled. Not sure if there is a better place for this check.
Another option could be to explicitly call this function in the app.php file.

Reply:
If it will checked only when app is enabled `user_webdavauth` can be enabled after the `user_ldap` and nobody will know that apps are incompatible

Reply:
moved into app.php